### DISCUSSION TITLE
Deluge+Synology+Flexget

### DISCUSSION THREAD

**Discussion by mar-cun** [Flexget/Flexget#3726] [Q&A]
I have Deluge and Flexget both installed as SynoCommunity apps on my Synology NAS, but when I try to invoke Deluge in my config it gives me an error that "deluge-client" is missing. Do I need to tell it how to find deluge somehow? I tried defining the config_path variable.

  TestingDeluge:
    rss: http://www.***/
    accept_all: yes
    deluge:
      host: 192.168.1.100
      config_path: /volume1/@appdata/deluge/
      port: 58846
      username: '{? deluge.username ?}'
      password: '{? deluge.password ?}'
      label: anime

**Comment by gazpachoking** [comment]
You need to `pip install deluge-client` from the environment that flexget is installed in. It's a needed python dependency to talk to deluge.

**Reply by mar-cun** [reply]
Any idea how one would do that on a Synology NAS? I can SSH in no problem, just not quite sure of the commands to install the deluge client into the correct instance of python that is running on the server (in this case python 3.10), as there are multiple versions. Also, the last time I attempted to install a component using sudo it broke the installation and I had to reinstall it and all the associated apps.

**Reply by mar-cun** [reply]
If I just try "pip install deluge-client" from putty I get:

Traceback (most recent call last):
  File "/bin/pip", line 5, in <module>
    from pip._internal.cli.main import main
ModuleNotFoundError: No module named 'pip._internal'

**Comment by girtsl** [comment]
My recommendations from personal experience:
- Don't mess with DSM and use docker instead. Set each app in their own docker container, the images are readily available.
- Set up vpn container and make it the network for your deluge container.
- You'll need to start the containers from shell (SSH) because in DSM not everything is configurable from GUI.
- Create a docker-compose file.
